WordPress/gutenberg
# [RNMobile] Add integration tests to cover Drag & Drop functionality

## What?


Add integration tests for both `Draggable` and `BlockDraggable` components, as part of the Drag & Drop functionality.

**NOTE:** This PR requires the changes from https://github.com/WordPress/gutenberg/pull/41337 because the integration tests use the new testing APIs provided by the `react-native-gesture-handler` library.

## Why?


Addresses most of the test cases outlined in https://github.com/wordpress-mobile/gutenberg-mobile/issues/4881.

## How?


This PR covers the following test cases:
- Drag mode
    - Text blocks
        - When the block is unselected, long-pressing over the content enables drag mode.
        - When the block is selected, long-pressing over the content enables the drag mode.
        - When the block is selected and text is being edited, long-pressing over the content doesn't enable the drag mode.
        - When the block is selected and text is being edited, long-pressing over other blocks enables the drag mode and disables text editing.
    - Media blocks
        - When the block is unselected, long-pressing over the content enables drag mode.
        - When the block is selected, long-pressing over the content enables the drag mode.
    - Nested blocks
        - When the block is unselected, long-pressing over the content enables drag mode.
        - When the block is selected, long-pressing over the content enables the drag mode.
        - When a nested block is selected, long-pressing over the content enables the drag mode.
    - Other blocks
        - When the block is unselected, long-pressing over the content enables drag mode.
        - When the block is selected, long-pressing over the content enables the drag mode.
- Dropping a block re-orders the block list.

The integration tests required applying different modifications to the following parts:
- Update the `AztecView` component mock to enable the logic related to `AztecInputState`.
- Add `testID` props to Draggable components to allow querying dragging elements on tests, specifically in:
    - `BlockDraggableChip` component
    - `BlockDraggable` component
    - `Draggable` component
    - `DraggableTrigger` component
- Add test helpers to simplify the use of the testing APIs provided by the `react-native-gesture-handler` library.

The Drag & Drop functionality relies heavily on the `react-native-reanimated` library, this introduced a new complexity because it uses timers for most of its logic, like managing the shared values and animation reaction hooks. For this reason, additional helpers have been added to simplify the testing logic.

## Testing Instructions




1. Run `npm run native test` and observe that all tests pass.
2. Check that mobile unit tests CI job succeeds.

## Screenshots or screencast 
N/A
